lib/address.js

```diff
@@ -112,6 +112,8 @@ Address.prototype._classifyArguments = function(data, network, type) {
     return Address._transformScript(data, network);
   } else if (typeof(data) === 'string') {
     return Address._transformString(data, network, type);
+  } else if (_.isObject(data) && data.network && data.hash && data.type) {
```

Comment:
I'd rather only take this path if the properties are set. Maybe in the future there are other structures for objects that we can accept.